src/io/flutter/editor/WidgetIndentGuideDescriptor.java

```diff
@@ -22,25 +27,80 @@
  * edited as there will be a slight delay before new analysis data is available.
  */
 public class WidgetIndentGuideDescriptor {
+  public static class WidgetPropertyDescriptor {
+    private RangeMarker marker;
+    private final FlutterOutlineAttribute attribute;
+
+    WidgetPropertyDescriptor(FlutterOutlineAttribute attribute) {
+      this.attribute = attribute;
+    }
+
+    public String getName() { return attribute.getName();}
+
+    public FlutterOutlineAttribute getAttribute() {
+      return attribute;
+    }
+
+    public int getEndOffset() {
+      if (marker == null) {
+        final Location location = attribute.getValueLocation();
+        final int startOffset = attribute.getValueLocation().getOffset();
+        final int endOffset = startOffset + location.getLength();
+        return endOffset;
+      }
+      return marker.getEndOffset();
+    }
+
+    public void track(Document document) {
+      if (marker != null) {
+        // TODO(jacobr): it does indicate a bit of a logic bug if we are calling this method twice.
+        assert (marker.getDocument() == document);
+        return;
+      }
+
+      // Create a range marker that goes from the start of the indent for the line
+      // to the column of the actual entity.
+      final int docLength = document.getTextLength();
+      final Location location = attribute.getValueLocation();
+      final int startOffset = Math.min(attribute.getValueLocation().getOffset(), docLength);
+      final int endOffset = Math.min(startOffset + location.getLength(), docLength);
+
+      marker = document.createRangeMarker(startOffset, endOffset);
+//      nodeStartingWord = OutlineLocation.getCurrentWord(document, nameExpression);
```

Comment:
remove this?

Reply:
removed. the helper class for tracking a location now has that functionality.